schematic/store/synapse.py

```diff
@@ -693,28 +706,35 @@ def getFilesInStorageDataset(
         """
         file_list = []
 
-        # Identify all folders nested under the dataset folder
-        folders = synapseutils.walk(self.syn, datasetId, includeTypes=["folder"])
-
-        # Getting the files directly under the dataset will be the beginning of the query
-        dataset_clause = f"parentId='{datasetId}' "
-
-        # The query will also be ammended to include everything containted in all the subdirectories of the dataset
-        for subfolder, _, path in folders:
-            dataset_clause += f"OR parentId='{subfolder[1]}' "
-        dataset_clause = f"({dataset_clause})"
+        # Get path to dataset folder from fileview to avoid building a new fileview and walking to determine folders and files within
+        child_path = self.storageFileviewTable.loc[
+            self.storageFileviewTable["parentId"] == datasetId, "path"
+        ][0]
+        parent = child_path.split("/")[0]
```

Comment:
Should we guard against index errors if the .loc returns an empty list? It would occur if the datasetId is not in the file view table. I'd think we could add a guard condition and raise an exception is not found.

Reply:
Will doing a split on "/" and taking the first of the list work for nested datasets? I was thinking of this case where a dataset is at

Root/sub/dir/dataset

I'd think it would return "Root" in this case, which isn't really looking at the path we want to get files at.

Should this instead be to split and use everything but the last of the split list, then join them back with a "/"?

Reply:
Ah, good catch @BryanFauble , yes, indeed that is the case.  We want to remove just the last element after the split.